### PR TITLE
Add GTKWave Flatpak port

### PR DESCRIPTION
**Pull request by aklsh**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by aklsh** [comment]
bot, build net.sourceforge.GTKWave

File: net.sourceforge.GTKWave.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  - --filesystem=home
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --socket=wayland
+modules:
+  - name: tcl
+    sources:
+      - type: archive
+        url: https://onboardcloud.dl.sourceforge.net/project/tcl/Tcl/8.6.12/tcl8.6.12-src.tar.gz
```
Comment (hfiguiere) [inline net.sourceforge.GTKWave.yml:20]:
```suggestion
        url: https://downloads.sourceforge.net/project/tcl/Tcl/8.6.12/tcl8.6.12-src.tar.gz
```

File: net.sourceforge.GTKWave.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        sha256: 26c995dd0f167e48b11961d891ee555f680c175f7173ff8cb829f4ebcde4c1a6
+    subdir: "unix"
+    post-install:
+      - chmod 755 /app/lib/libtcl*.so
+  - name: tk
+    sources:
+      - type: archive
+        url: https://onboardcloud.dl.sourceforge.net/project/tcl/Tcl/8.6.12/tk8.6.12-src.tar.gz
```
Comment (hfiguiere) [inline net.sourceforge.GTKWave.yml:28]:
```suggestion
        url: https://downloads.sourceforge.net/project/tcl/Tcl/8.6.12/tk8.6.12-src.tar.gz
```

File: net.sourceforge.GTKWave.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+sdk: org.gnome.Sdk
+command: gtkwave
+rename-icon: gtkwave
+rename-desktop-file: gtkwave.desktop
+rename-appdata-file: gtkwave.appdata.xml
+finish-args:
+  - --device=dri
+  - --filesystem=home
+  - --share=ipc
+  - --socket=fallback-x11
```
Comment (hfiguiere) [inline net.sourceforge.GTKWave.yml:13]: Have you tested this on a wayland session, since you build Tk.

**Comment by hfiguiere** [comment]
bot, build net.sourceforge.GTKWave

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: net.sourceforge.GTKWave.yml (hunk 1 above)
Comment (aklsh) [reply net.sourceforge.GTKWave.yml]: Done!

File: net.sourceforge.GTKWave.yml (hunk 2 above)
Comment (aklsh) [reply net.sourceforge.GTKWave.yml]: Done!

File: net.sourceforge.GTKWave.yml (hunk 3 above)
Comment (aklsh) [reply net.sourceforge.GTKWave.yml:13]: Yes. I'm able to open the app and use it normally without it crashing.

Comment (hfiguiere) [reply net.sourceforge.GTKWave.yml:13]: so what is Tk for?

**Comment by hfiguiere** [comment]
There is no appstream file to be found.

File: net.sourceforge.GTKWave.yml (hunk 3 above)
Comment (aklsh) [reply net.sourceforge.GTKWave.yml:13]: I wasn't able to build the application without it.

**Comment by aklsh** [comment]
> There is no appstream file to be found.

I've added the appstream file now. It wasn't being installed in the sandbox when building the application due to naming scheme and missing tags - have updated it now :)

**Comment by aklsh** [comment]
bot, build net.sourceforge.GTKWave

**Comment by hfiguiere** [comment]
still not found

File: net.sourceforge.GTKWave.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    config-opts:
+      - --enable-gtk3
+      - --enable-manymarkers
+      - --disable-dependency-tracking
+      - --disable-mime-update
+  - name: appdata
+    buildsystem: simple
+    build-commands:
+      - install -Dm644 net.sourceforge.GTKWave.appdata.xml /app/share/appdata/net.sourceforge.GTKWave.appdata.xml
+    sources:
```
Comment (hfiguiere) [inline net.sourceforge.GTKWave.yml:47]:
```suggestion
  - name: gtkwave
    config-opts:
      - --enable-gtk3
      - --enable-manymarkers
      - --disable-dependency-tracking
      - --disable-mime-update
    post-install:
      - install -Dm644 net.sourceforge.GTKWave.appdata.xml /app/share/appdata/net.sourceforge.GTKWave.appdata.xml
    sources:
      - type: archive
        url: http://gtkwave.sourceforge.net/gtkwave-gtk3-3.3.111.tar.gz
        sha256: b4e1c2f718fb3c4b4f412a220876b9da599fe11745fb6f7eb4aed107b1106233
```

File: net.sourceforge.GTKWave.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
+app-id: net.sourceforge.GTKWave
+runtime: org.gnome.Platform
+runtime-version: '42'
+sdk: org.gnome.Sdk
+command: gtkwave
+rename-icon: gtkwave
+rename-desktop-file: gtkwave.desktop
+rename-appdata-file: gtkwave.appdata.xml
```
Comment (hfiguiere) [inline net.sourceforge.GTKWave.yml:8]: if you have it upstream (as the rename seem to indicate) there is no reason to have a copy here.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: net.sourceforge.GTKWave.yml (hunk 5 above)
Comment (aklsh) [reply net.sourceforge.GTKWave.yml:8]: I tried with that initially, but it looks like there are some necessary tags missing that are required for flatpak. So, I've created a copy of it locally.

**Comment by aklsh** [comment]
bot, build net.sourceforge.GTKWave

File: net.sourceforge.GTKWave.yml (hunk 5 above)
Comment (hfiguiere) [reply net.sourceforge.GTKWave.yml]: usually what is done is patches. So they get submitted upstream too.

**Comment by hfiguiere** [comment]
the id in the appdata file is incorrect.

File: net.sourceforge.GTKWave.metainfo.xml [hunk 6]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2014 Tony Bybell <bybell@rocketmail.com> -->
+<component type="desktop">
+  <id>net.sourceforge.gtkwave</id>
```
Comment (hfiguiere) [inline net.sourceforge.GTKWave.metainfo.xml:4]: this is case sensitive, therefore incorrect.

Please use the appdata file from upstream and patch as necessary. (and if you are upstream make sure the patch is in the next release)

File: net.sourceforge.GTKWave.metainfo.xml [hunk 7]
```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2014 Tony Bybell <bybell@rocketmail.com> -->
+<component type="desktop">
+  <id>net.sourceforge.gtkwave.desktop</id>
```
Comment (hfiguiere) [inline net.sourceforge.GTKWave.metainfo.xml:4]:
```suggestion
  <id>net.sourceforge.GTKWave</id>
```

`.desktop` should never be here. it's a deprecated thing.

Comment (aklsh) [reply net.sourceforge.GTKWave.metainfo.xml:4]: Oh sorry. I was looking at the logs and I saw some .desktop comparison in the regex and thought it was necessary.

Comment (hfiguiere) [reply net.sourceforge.GTKWave.metainfo.xml]: the regexp tolerate the deprecated format.

**Comment by hfiguiere** [comment]
as I said a patch would be better.

**Comment by aklsh** [comment]
> as I said a patch would be better.

Sure. I shall do that. Didn't see your previous comment

**Comment by aklsh** [comment]
Is there a way in which I will be able to test the build process locally before I start a build on flathub servers so that I can catch these errors sooner? I'm totally new to this and tend to make a lot of mistakes.

**Comment by aklsh** [comment]
> Is there a way in which I will be able to test the build process locally before I start a build on flathub servers so that I can catch these errors sooner? I'm totally new to this and tend to make a lot of mistakes.

Something like a shell script or docker container?

**Comment by aklsh** [comment]
bot, build net.sourceforge.GTKWave

**Comment by nedrichards** [comment]
bot, build net.sourceforge.GTKWave

**Comment by aklsh** [comment]
Have created a PR upstream with the appdata patch - gtkwave/gtkwave#142

The build works!! 🥳

File: net.sourceforge.GTKWave.yml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+      - --enable-manymarkers
+      - --disable-dependency-tracking
+      - --disable-mime-update
+    subdir: "gtkwave3-gtk3"
+    post-install:
+      - install -Dm644 share/appdata/gtkwave.appdata.xml /app/share/metainfo/net.sourceforge.GTKWave.metainfo.xml
+    sources:
+      - type: git
+        url: http://github.com/gtkwave/gtkwave
+        branch: master
```
Comment (nedrichards) [inline net.sourceforge.GTKWave.yml:44]: this needs to be a specific commit rather than just a branch that can keep changing. other than that lgtm

**Comment by hfiguiere** [comment]
bot, build io.github.gtkwave.GTKWave

**Pull request closed without merging**